pkg/cloud/vsphere/vsphere_cloud_config/config.go

```diff
@@ -0,0 +1,45 @@
+package vsphere_cloud_config
+
+import (
+	"fmt"
+
+	"k8s.io/klog/v2"
+
+	"gopkg.in/yaml.v2"
+)
+
+// ReadConfig parses vSphere cloud-config file and returns CPIConfig structure
+// Accepts both YAML and INI formats as input.
+// YAML format takes precedence, in case parsing YAML is not successful function falls back to the legacy INI format.
+// Unlike 'cloud-provider-vsphere' version of a similar function, this does ignore environment variables.
+func ReadConfig(config []byte) (*CPIConfig, error) {
+	if len(config) == 0 {
+		return nil, fmt.Errorf("vSphere config is empty")
```

Comment:
Nit: Without a format string, should use `errors.New`
```suggestion
		return nil, errors.New("vSphere config is empty")
```